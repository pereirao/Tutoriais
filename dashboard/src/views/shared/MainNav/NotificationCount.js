// @flow

import React from 'react';
import {connect} from 'react-redux';
import {notification_selectors} from '../../../business/notification';
import style_sheet from './MainNav.module.css';

const NotificationCountSTP = state => ({notice_count: notification_selectors.totalCount(state)});
const connectNotificationCount = connect(NotificationCountSTP);
type NotificationCountProps = { notice_count: number };
export const NotificationCount = ({notice_count}: NotificationCountProps) => {
  if (!notice_count) return null;
  const bubble_width = (notice_count.toString().length * 8) + 12;
  return (
    <div className={style_sheet.noticeCount} style={{width: bubble_width}}>{notice_count}</div>
  );
};

export default connectNotificationCount(NotificationCount);
