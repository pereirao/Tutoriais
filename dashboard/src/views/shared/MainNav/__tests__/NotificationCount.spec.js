import React from 'react';

import { NotificationCount } from '../NotificationCount';

it('renders 1 digit correctly', () => {
  expect(shallow(<NotificationCount notice_count={1} />)).toMatchSnapshot();
});

it('renders 2 digits correctly', () => {
  expect(shallow(<NotificationCount notice_count={10} />)).toMatchSnapshot();
});

it('renders 3 digits correctly', () => {
  expect(shallow(<NotificationCount notice_count={101} />)).toMatchSnapshot();
});
