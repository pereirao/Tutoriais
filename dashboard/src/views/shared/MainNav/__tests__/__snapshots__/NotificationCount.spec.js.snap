// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1 digit correctly 1`] = `
<div
  className="noticeCount"
  style={
    Object {
      "width": 20,
    }
  }
>
  1
</div>
`;

exports[`renders 2 digits correctly 1`] = `
<div
  className="noticeCount"
  style={
    Object {
      "width": 28,
    }
  }
>
  10
</div>
`;

exports[`renders 3 digits correctly 1`] = `
<div
  className="noticeCount"
  style={
    Object {
      "width": 36,
    }
  }
>
  101
</div>
`;
